### ISSUE TITLE
status: progress 21 of 18 done

### ISSUE THREAD

**Issue by tiborsimko** [reanahub/reana-client#391]
For some dynamic workflows, the number of steps/jobs is not known in advance. Some steps/jobs are created by the workflow say in scatter-gather situations as the workflow is running.This leads to situations such as: 

- status=finished, steps=21/18

which may be hard to interpret by the users. 

We could:

- document this better
- or change the way how the progress is calculated

The former is easy, the latter is the focus of this issue.

Basically, we have to be ready that the total number of steps for a workflow can change as the workflow runs. 

See the past discussion in https://github.com/reanahub/reana-workflow-engine-serial/pull/14 about 
N1 planned, N2 succeeded, N3 failed, N4 submitted, N total jobs for a workflow.

We can revive the issue based on those musings.
